src/Angor.Test/WalletOperationsTest.cs

```diff
@@ -183,10 +202,249 @@ public void AddInputsAndSignTransaction()
+
+        var walletOperations = new WalletOperations(null, mockHdOperations.Object, null, null);
+
+        // Act
+        var (coins, keys) = walletOperations.GetUnspentOutputsForTransaction(walletWords, utxos);
+
+        // Assert
+        Assert.Single(coins);
+        Assert.Single(keys);
+        Assert.Equal((uint)0, coins[0].Outpoint.N);
+        Assert.Equal(1500000, coins[0].Amount.Satoshi);
+        Assert.Equal(expectedExtKey.Derive(new KeyPath("m/0/0")).PrivateKey, keys[0]);
+    }
+
+    [Fact]
+    public void CalculateTransactionFee_WithMultipleScenarios()
+    {
+        // Arrange common elements
+        var mockNetworkConfiguration = new Mock<INetworkConfiguration>();
+        var mockIndexerService = new Mock<IIndexerService>();
+        var mockHdOperations = new Mock<IHdOperations>();
+        var mockLogger = new Mock<ILogger<WalletOperations>>();
+        var network = _networkConfiguration.Object.GetNetwork();
+        mockNetworkConfiguration.Setup(x => x.GetNetwork()).Returns(network);
+
+        var walletOperations = new WalletOperations(mockIndexerService.Object, mockHdOperations.Object, mockLogger.Object, mockNetworkConfiguration.Object);
+
+        var words = new WalletWords { Words = "suspect lesson reduce catalog melt lucky decade harvest plastic output hello panel", Passphrase = "" };
+        var address = "tb1qeu7wvxjg7ft4fzngsdxmv0pphdux2uthq4z679";
+        var scriptHex = "0014b7d165bb8b25f567f05c57d3b484159582ac2827";  
+        var accountInfo = new AccountInfo(); 
+        long feeRate = 10; 
+
+        // Scenario 1: Sufficient funds
+        var sendInfoSufficientFunds = new SendInfo
+        {
+            SendToAddress = "tb1qw4vvm955kq5vrnx48m3x6kq8rlpgcauzzx63sr",
+            ChangeAddress = "tb1qw4vvm955kq5vrnx48m3x6kq8rlpgcauzzx63sr",
+            SendAmount = 0.0001m,  // Lower amount for successful fee calculation
+            SendUtxos = new Dictionary<string, UtxoDataWithPath>
+        {
+            {
+                "key", new UtxoDataWithPath
+                {
+                    UtxoData = new UtxoData
+                    {
+                        value = 150000000, // Sufficient to cover the send amount and estimated fees
+                        address = address,
+                        scriptHex = scriptHex,
+                        outpoint = new Outpoint("0000000000000000000000000000000000000000000000000000000000000000", 0),
+                        blockIndex = 1,
+                        PendingSpent = false
+                    },
+                    HdPath = "m/0/0"
+                }
+            }
+        }
+        };
+
+        // Act & Assert for sufficient funds
+        var calculatedFeeSufficient = walletOperations.CalculateTransactionFee(sendInfoSufficientFunds, accountInfo, feeRate);
+        Assert.True(calculatedFeeSufficient > 0);
+        Assert.Equal(0.00000001m, calculatedFeeSufficient);  // Assuming an expected fee for validation
```

Comment:
this is fine, the fee is exactly the same every time because the size of the trx is fixed and feerate is fixed, so it is fine to keep it.